src/JustEat.StatsD/ConnectedSocketPool.cs

```diff
@@ -7,22 +8,35 @@ namespace JustEat.StatsD
     internal sealed class ConnectedSocketPool : IDisposable
     {
         private bool _disposed;
-        private readonly SimpleObjectPool<Socket> _pool;
+        private readonly ConcurrentBag<Socket> _pool = new ConcurrentBag<Socket>();
+
+        public IPEndPoint IpEndPoint { get; }
 
         public ConnectedSocketPool(IPEndPoint ipEndPoint)
```

Comment:
Maybe we should make `count` a parameter and have the caller specify the value? (which would happen to be `Environment.ProcessorCount`)